config/config.go

```diff
@@ -0,0 +1,141 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+	"github.com/mcuadros/go-defaults"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+type Loader struct {
+	v *viper.Viper
+}
+
+type LoaderOption func(*Loader)
+
+func WithViper(in *viper.Viper) LoaderOption {
+	return func(l *Loader) {
+		l.v = in
+	}
+}
+
+func WithConfigName(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetConfigName(in)
+	}
+}
+
+func AddConfigPath(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.AddConfigPath(in)
+	}
+}
+
+func WithConfigType(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetConfigType(in)
+	}
+}
+
+func WithEnvKeyReplacer(old string, new string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetEnvKeyReplacer(strings.NewReplacer(old, new))
+	}
+}
+
+func NewLoader(options ...LoaderOption) *Loader {
+	loader := &Loader{
+		v: getViperWithDefaults(),
+	}
+
+	for _, option := range options {
+		option(loader)
+	}
+	return loader
+}
+
+// Load loads configuration into the given mapstructure (https://github.com/mitchellh/mapstructure)
+// from a config.yaml file and overrides with any values set in env variables
+func (l *Loader) Load(config interface{}) {
+	verifyParamIsPtrToStructElsePanic(config)
+
+	l.v.AutomaticEnv()
+
+	err := l.v.ReadInConfig()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Even for CLIs this fmt will print config file not found. Either we can allow creating a loader without config file or optional config file. Or maybe we can avoid printing and only return the error. So clients can decide if this maters to them or not. 

e..g 
```
if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
  return c, err
}

Thoughts?

Reply:
we can avoid printing it

Reply:
fixed